scripts/download_technique_animations.py

```diff
@@ -8,67 +8,126 @@
 If the file names have changed from the wiki, the technique
 json files will need to be updated as well.
 """
-from urllib.parse import urljoin
 import os.path
+import pathlib
+import tempfile
+from urllib.parse import urljoin
 
 import requests
 from PIL import Image
 from lxml import html
 
 FRAME_SIZE = 64
-WIKI_URL = "https://wiki.tuxemon.org/index.php?title=Category:Technique_Animation"
+WIKI_URL = "https://wiki.tuxemon.org"
+
+TUXEMON_ROOT_DIR = pathlib.Path(__file__).resolve().parent.parent
+ANIMATION_DIR = TUXEMON_ROOT_DIR.joinpath(pathlib.Path("mods/tuxemon/animations/technique"))
+CREDITS_TEMPLATE = '* ["{animation_name}"]({animation_url}) {credits_text}'
+CREDITS_FILENAME = "TECHNIQUE_ANIMATION_CREDITS.md"
+
+print("Tuxemon project root dir:", TUXEMON_ROOT_DIR)
+print("Technique animation dir:", ANIMATION_DIR)
 
-# TODO: Is the return here needed or should be boolean to indicate success/fail?
-def download_bytes(url: str, filename: str) -> str:
-    """Downloads a stream of bytes from a given URL to a file defined by 'filename'."""
+
+def download_bytes(url: str, filepath: str) -> None:
+    """Downloads a stream of bytes from a given URL to a file path."""
     req = requests.get(url)
-    with open(filename, "wb") as fp:
+    with open(filepath, "wb") as fp:
         for chunk in req.iter_content(chunk_size=1024):
             if chunk:
                 fp.write(chunk)
-    return filename
 
 
-def process_filename(filename: str) -> str:
+def process_filename(filepath: str) -> str:
     """Extract base filename from an animation file path."""
-    cleaned_filename = os.path.splitext(filename)[0]
+    cleaned_filename = os.path.splitext(os.path.basename(filepath))[0]
     cleaned_filename = cleaned_filename.strip("0123456789_")
     cleaned_filename = cleaned_filename.lower()
     return cleaned_filename
 
 
-def process_gif(filename: str) -> None:
+def process_animation_name(filename: str) -> str:
+    """Convert filename of format 'file_name' to a capitalized 'File Name' animation name/title."""
+    animation_name = " ".join([name_part.capitalize() for name_part in filename.split("_")])
+    return animation_name
+
+
+def download_animation_credits(gif_page_url: str) -> str:
+    """Extract credits text from a GIF animation subpage, from the 'Comments' section."""
+    gif_filename: str = gif_page_url.split("File:")[-1]
```

Comment:
Do we need the typehints for the variables? Should be obvious to mypy and Devs that these are strings.

Reply:
I can't find an authoritarian source for this, but I wouldn't use a variable annotation unless I had to.

https://stackoverflow.com/a/51214849

Reply:
I usually overdo it, so it's extra clear, but I will remove it from places where it doesn't belong.